Mathlib/Order/Interval/Set/IsoIoo.lean

```diff
@@ -20,7 +20,8 @@ open Set
 We consider the actual implementation to be a "black box", so it is irreducible.
 -/
 @[irreducible]
-def orderIsoIooNegOneOne (k : Type*) [LinearOrderedField k] : k ≃o Ioo (-1 : k) 1 := by
+def orderIsoIooNegOneOne (k : Type*) [Field k] [LinearOrder k] [IsStrictOrderedRing k] :
+    k ≃o Ioo (-1 : k) 1 := by
```

Comment:
buzzard note to self: I have reviewed up until here. I have a two-hour committee meeting to go to right now though, I will start again from this point afterwards but I'll finish the review so people have the chance to look at the issues which I've raised so far.